src/Controller/PageAdminController.php

```diff
@@ -265,4 +289,13 @@ private function setFormTheme(FormView $formView, $theme)
 
         $twig->getRuntime(FormRenderer::class)->setTheme($formView, $theme);
     }
+
+    /**
+     * NEXT_MAJOR: Check if it was added in SonataAdminBundle, if yes remove this method!
+     * @internal
+     */
+    protected function getAdmin(): AdminInterface
```

Comment:
@VincentLanglet is it fine add this method into https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Controller/CRUDController.php ? it can help to mock the admin.

Reply:
I don't see a real benefit.

Adding code for tests is not a good pattern. And you can mock the adminFetcher
```
$adminFetcher = $this->container->get('sonata.admin.request.fetcher');
        \assert($adminFetcher instanceof AdminFetcherInterface);

        /** @var AdminInterface<T> $admin */
        $admin = $adminFetcher->get($request);
        $this->admin = $admin;
```

Reply:
@VincentLanglet , But I can not mock `$this->admin` using your example! it does not work for me!

Note: if you look into this test: https://github.com/sonata-project/SonataPageBundle/pull/1418/files#diff-2486e8a2bb13c5d093018e6058f6d13246f449650d15ea1bc41d2f3b7136039f, I'm extending from `TestCase`

Reply:
You can mock the adminFetcher return value, like we did in SonataAdmin https://github.com/sonata-project/SonataAdminBundle/blob/4.x/tests/Controller/CRUDControllerTest.php#L154-L157

Reply:
> You can mock the adminFetcher return value, like we did in SonataAdmin https://github.com/sonata-project/SonataAdminBundle/blob/4.x/tests/Controller/CRUDControllerTest.php#L154-L157

I can create the mock like this, but the question is, how can I put this mock in `$this->admin` in PageAdminController


Maybe it's really clear and I can't see how to do this 😄

Reply:
Ok, tomorrow I'll try again follow your example :)

Reply:
Hey @VincentLanglet 

I tried to use the setUp that you mentioned, But I'm getting this error

```
RuntimeException : There is no `_sonata_admin` defined for the controller `Sonata\PageBundle\Controller\PageAdminController` and the current route ``.
```

any idea why?

Reply:
I removed the `getAdmin` method, But I used Reflection is it fine for you?